docs/frameworks/symfony.md

```diff
@@ -133,6 +133,29 @@ The secrets (e.g. database passwords) must however not be committed in this file
 
 To learn more about all this, read the [environment variables documentation](/docs/environment/variables.md).
 
+## Trust API Gateway
+
+When hosting your site on Lambda, API Gateway will act as a proxy between the client and your function. 
+
+By default, Symfony doesn't trust proxies for security reasons, but it's safe to do it when using API Gateway and
+Lambda. 
+
+This is needed because otherwise, Symfony will not be able to generate URLs properly.
+
+You should add the following lines to `config/packages/framework.yaml`
+
+```yaml
+framework:
+    # trust the remote address because API Gateway has no fixed IP or CIDR range that we can target
+    trusted_proxies: '127.0.0.1,REMOTE_ADDR'
+    # trust "X-Forwarded-*" headers coming from API Gateway (note that it doesn't send X-Forwarded-Host)
```

Comment:
API Gateway doesn't send that header, but it's possible to send it manually when using CloudFront (which is something recommended in Bref's documentation, also you can see an example [here](https://github.com/brefphp/bref/issues/535#issuecomment-577158098)).

So I would recommend allowing `X-Forwarded-Host` too to be forward-compatible, in case the user wants to use CloudFront and sets up that header.

Reply:
IMHO this is a bad idea to enable it by default because we're not sure it will be configured and that can be a security problem.

From https://symfony.com/doc/current/deployment/proxies.html#solution-settrustedproxies
> Enabling the Request::HEADER_X_FORWARDED_HOST option exposes the application to [HTTP Host header attacks](https://www.skeletonscribe.net/2013/05/practical-http-host-header-attacks.html). Make sure the proxy really sends an x-forwarded-host header.

So this could expose the application to host poisoning attacks if they don't configure CloudFront to send that header. But maybe we should add a note to say that it should be enabled if CloudFront is configured properly ?

Reply:
Very good point.

> But maybe we should add a note to say that it should be enabled if CloudFront is configured properly ?

👍  let's go with "disabled by default" (in the docs). Maybe a note like this:

> If you set the `X-Forwarded-Host` manually, for example in CloudFront configuration, you can also whitelist that header.

Reply:
Updated to include X-Forwarded-Host recommendations !